cpp/src/arrow/ArrowConfig.cmake.in

```diff
@@ -172,3 +172,16 @@ endmacro()
 arrow_keep_backward_compatibility(Arrow arrow)
 
 check_required_components(Arrow)
+
+function(arrow_show_details PACKAGE_NAME)
+  if(NOT ${PACKAGE_NAME}_FIND_QUIETLY)
+    string(TOUPPER ${PACKAGE_NAME} package_name_upper)
+    message(STATUS "${PACKAGE_NAME} version: ${${package_name_upper}_VERSION}")
```

Comment:
Can we use `${PACKAGE_NAME}_VERSION` here?
`ARROW_VERSION` is defined but `ARROW_DATASET_VERSION` and so on aren't defined.